src/Codec/StringCodec.php

```diff
@@ -118,6 +118,10 @@ protected function getBytes(string $encodedUuid): string
             $encodedUuid
         );
 
+        if (strpos($parsedUuid, '0x') === 0) {
+            $parsedUuid = substr($parsedUuid, 2);
+        }
```

Comment:
I think this belongs in `UuidFactory::fromHexadecimal()`, before passing the value to the codec for decoding.

Reply:
Reason for adding it here is to allow the hex string to be used via this method. See the new test cases.

Reply:
I think this should throw an exception, since it's not a valid string UUID:

``` php
Uuid::fromString('0x1EA78DEB37CE625E8F1A025041000001')
```

The main problem here is what to do if someone configures the library to use a different codec. Then, neither `fromString()` nor `fromHexadecimal()` will work as expected.

Reply:
Makes perfect sense. Will update accordingly.